### PR TITLE
Add support for MinMon data

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
## Description

Extend `mon_data_space.py` to support MinMon data sources, which are 1 dimensional arrays, and have no header or footer (as Fortran binary files do).  Add `Iteration` as an available data dimension.

Add a mechanism to `mon_data_space.py` to zero out data for missing files.  Time series data occasionally has missing data files and in these cases the desired plot is a cycle time with zero/missing data. 

Lastly extend `transforms/arithmetic.py` by adding log() transform.  This is needed by MinMon gnorm plots.

## Dependencies

None

## Impact

None

### CODE REVIEW

**Comment by EdwardSafford-NOAA** [comment]
@CoryMartin-NOAA @kevindougherty-noaa @ADCollard please review if you have a moment.  Thanks.

File: src/eva/data/mon_data_space.py [hunk 1]
```diff
@@ -291,7 +292,8 @@ def read_ieee(self, file_name, coords, dims, ndims_used, dims_arr, nvars, vars,
 
         # find cycle time in filename and create cycle_tm as datetime object
         cycle_tm = None
-        cycstrs = file_name.split('.')
+        fname = file_name.replace('/', '.')
+        cycstrs = fname.split('.')
```
Comment (kevindougherty-noaa) [inline src/eva/data/mon_data_space.py]: If you don't need the variable `fname` anywhere else, you can combine these two lines into:
`cycstrs = file_name.replace('/', '.').split('.')`

However, it doesn't need to be changed.

**Review by kevindougherty-noaa — APPROVED** [review]
Looks good to me. Just had a minor comment that doesn't need to be changed.

File: src/eva/data/mon_data_space.py [hunk 2]
```diff
@@ -300,52 +302,66 @@ def read_ieee(self, file_name, coords, dims, ndims_used, dims_arr, nvars, vars,
 
         filename = os.path.join(file_path, file_name) if file_path else file_name
 
+        load_data = True
         if not os.path.isfile(filename):
```
Comment (CoryMartin-NOAA) [inline src/eva/data/mon_data_space.py]: nit pick, but I would prefer the `if` be flipped and not do an `if not` and then `else` but rather `if` then `else`

File: src/eva/data/mon_data_space.py [hunk 3]
```diff
@@ -300,52 +302,66 @@ def read_ieee(self, file_name, coords, dims, ndims_used, dims_arr, nvars, vars,
...
-        if ndims_used == 1:
-            rtn_array = np.empty((0, dims[dims_arr[0]]), float)
+        if ndims_used == 1:		# MinMon
+            rtn_array = np.empty((0, dims[dims_arr[0]]), dtype=float)
+            if not load_data:
+                zarray = np.zeros((dims[dims_arr[0]]), float)
             dimensions = [dims[dims_arr[0]]]
 
-        if ndims_used == 2:
+        if ndims_used == 2:		# RadMon time, bcoef, bcor, OznMon time
```
Comment (CoryMartin-NOAA) [inline src/eva/data/mon_data_space.py:317]: this might break pycodestyle conventions?

File: src/eva/data/mon_data_space.py [hunk 4]
```diff
@@ -300,52 +302,66 @@ def read_ieee(self, file_name, coords, dims, ndims_used, dims_arr, nvars, vars,
...
             rtn_array = np.empty((0, dims[dims_arr[0]], dims[dims_arr[1]]), float)
+            if not load_data:
+                zarray = np.zeros((dims[dims_arr[0]], dims[dims_arr[1]]), float)
             dimensions = [dims[dims_arr[0]], dims[dims_arr[1]]]
 
-        if ndims_used == 3:
+        if ndims_used == 3:		# RadMon angle
```
Comment (CoryMartin-NOAA) [inline src/eva/data/mon_data_space.py:323]: same comment, I think that might be too many spaces (unless the linter doesn't complain then it's fine)

**Review by CoryMartin-NOAA — COMMENTED** [review]
Couple of minor things

File: src/eva/data/mon_data_space.py (hunk 3 above)
Comment (EdwardSafford-NOAA) [reply src/eva/data/mon_data_space.py:317]: No apparently not.  All tests passed.

File: src/eva/data/mon_data_space.py (hunk 4 above)
Comment (EdwardSafford-NOAA) [reply src/eva/data/mon_data_space.py:323]: As above linter is good with this.

File: src/eva/data/mon_data_space.py (hunk 3 above)
Comment (CoryMartin-NOAA) [reply src/eva/data/mon_data_space.py:317]: Ok, works for me.

File: src/eva/data/mon_data_space.py (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply src/eva/data/mon_data_space.py]: No problem.  Change made.

File: src/eva/data/mon_data_space.py (hunk 2 above)
Comment (EdwardSafford-NOAA) [reply src/eva/data/mon_data_space.py]: Fair point, change made.

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good, thanks!

**Pull request merged**